### ISSUE TITLE
Optimization and weight_decay on the final linear classifier

### ISSUE THREAD

**Issue by CodingTomo** [sun-hailong/LAMDA-PILOT#21]
Hi everyone,
Thank you all for your hard work! :)
I have a question regarding the management of optimization for the incremental linear classifier. In the following, I use the `FineTuning` approach as an example.

Here the definition of the optimizer when `self._cur_task != 0`
```
optimizer = optim.SGD(
                self._network.parameters(),
                lr=self.args["lrate"],
                momentum=0.9,
                weight_decay=self.args["weight_decay"],
            ) 
```

In the function `self._update_representation`, within the training loop and using the above optimizer, you execute the following lines:
```
logits = self._network(inputs)["logits"]
fake_targets = targets - self._known_classes
loss_clf = F.cross_entropy(
    logits[:, self._known_classes :], fake_targets
)

loss = loss_clf
optimizer.zero_grad()
loss.backward()
```

I understand that by using `logits[:, self._known_classes:]`, you consider only the new classes as desired in the cross-entropy loss (the others are already trained during the previous steps). However, in the optimizer, you include all the network parameters ( `self._network.parameters()`), particularly all the parameters of the linear head. Consequently, the weight decay (set to `0.0002`) should also influence the weights related to the old classes. Is this correct?

If this is true, I suppose this is not a desired behavior because the optimum of the weight regularization for weights that are not involved in the cross-entropy should be zero.

Many thanks!

**Comment by sun-hailong** [comment]
Hi @CodingTomo ,
Thanks for using PILOT. When training in the incremental stage, we only calculate the cross-entropy loss for the new classes, so the gradient will not propagate back to the parameters of the old classes, ensuring their parameters remain unaffected.
